File: implementation/scipamato/scipamato-public-entity/src/test/java/ch/difty/scipamato/public_/entity/CodeTest.java
package ch.difty.scipamato.public_.entity;

import static org.assertj.core.api.Assertions.assertThat;

import org.junit.Test;

import nl.jqno.equalsverifier.EqualsVerifier;
import nl.jqno.equalsverifier.Warning;

public class CodeTest extends PublicEntityTest<Code> {

    @Override
    protected Code newEntity() {
        return Code.builder()
            .codeClassId(1)
            .code("code")
            .langCode("lc")
            .name("name")
            .comment("comment")
            .sort(3)
            .build();
    }

    @Override
    protected void assertSpecificGetters() {
        assertThat(getEntity().getCodeClassId()).isEqualTo(1);
        assertThat(getEntity().getCode()).isEqualTo("code");
        assertThat(getEntity().getLangCode()).isEqualTo("lc");
        assertThat(getEntity().getName()).isEqualTo("name");
        assertThat(getEntity().getComment()).isEqualTo("comment");
        assertThat(getEntity().getSort()).isEqualTo(3);
    }

    @Override
    protected String getToString() {
        return "Code(codeClassId=1, code=code, langCode=lc, name=name, comment=comment, sort=3)";
    }

    @Override
    protected void verifyEquals() {
        EqualsVerifier.forClass(Code.class)
            .withRedefinedSuperclass()
            .withIgnoredFields(Code.CREATED, Code.MODIFIED)
            .suppress(Warning.STRICT_INHERITANCE, Warning.NONFINAL_FIELDS)
            .verify();
    }

    @Test
    public void displayValue() {
        assertThat(getEntity().getDisplayValue()).isEqualTo("name");
    }
}

File: public/public-entity/src/test/kotlin/ch/difty/scipamato/publ/entity/CodeTest.kt
package ch.difty.scipamato.publ.entity

import nl.jqno.equalsverifier.EqualsVerifier
import org.amshove.kluent.shouldBeEqualTo
import org.junit.jupiter.api.Test
import java.time.LocalDateTime

internal class CodeTest : PublicDbEntityTest<Code>() {

    override fun newEntity(): Code = Code(
        codeClassId = 1,
        code = "code",
        langCode = "lc",
        name = "name",
        comment = "comment",
        sort = 3,
    )

    override fun assertSpecificGetters() {
        entity.codeClassId shouldBeEqualTo 1
        entity.code shouldBeEqualTo "code"
        entity.langCode shouldBeEqualTo "lc"
        entity.name shouldBeEqualTo "name"
        entity.comment shouldBeEqualTo "comment"
        entity.sort shouldBeEqualTo 3
    }

    override fun verifyEquals() {
        EqualsVerifier.simple().forClass(Code::class.java).verify()
    }

    @Test
    fun displayValue() {
        entity.displayValue shouldBeEqualTo "name"
    }
}
